include/LightGBM/utils/array_args.h

```diff
@@ -103,7 +104,7 @@ class ArrayArgs {
     int j = end - 1;
     int p = i;
     int q = j;
-    if (start >= end) {
+    if (start >= end - 1) {
```

Comment:
@kruda Could you please extract corresponding to the #4272 part into a separate PR? This will allow to faster merge this fix and not wait for MVS review.
As a bonus, doing so also will remove you from "first contributor" group and help to iterate thought CI cycles faster🙂 .

Reply:
Thank you for suggestion! I've created a new pull request for that. https://github.com/microsoft/LightGBM/pull/4280